### DISCUSSION TITLE
RustDesk Keeps OLED Monitor on 24/7 (remote window makes local monitor never sleep)

### DISCUSSION THREAD

**Discussion by SilentConfig** [rustdesk/rustdesk#9466] [Q&A] [answered]
### Bug Description

Using RustDesk I am connecting to a headless machine.
The client machine I am connecting from is ignoring the windows power options which are set - to "turn off the display" (which is currently set to 5minutes).

I am connected to the host machine pretty much all the time my computer is on, which means my OLED monitor never runs it's maintenance procedure (This runs every time the monitor sleeps after being on for 4 hours or more).

However since the power settings are ignored this never happens and I'm worried about the lifespan of my display when using RustDesk.

### How to Reproduce

Connect to any host machine from a Windows 11 client & wait the amount of time set in power options until the monitor SHOULD sleep.

### Expected Behavior

Monitor should turn off on client machine when idle according to windows power plan.

### Operating system(s) on local side and remote side

Windows 11 -> Windows 11

### RustDesk Version(s) on local side and remote side

1.3.1

### Screenshots

![image](https://github.com/user-attachments/assets/e8af18d2-9188-45a7-a5cc-bf6625cc7019)
![image](https://github.com/user-attachments/assets/12622007-9f0b-48d1-b60d-8c1b5ed708a5)

**Comment by 21pages** [comment]
AnyDesk does the same during remote controlled.
![image](https://github.com/user-attachments/assets/9b8613be-8c7d-49df-a0fa-f1f294d5031f)

**Reply by SilentConfig** [reply]
Ok, But can this behaviour be changed? or do you have some suggestion so that using it doesn't damage my OLED display over time?

Thanks

**Reply by SilentConfig** [reply]
I appreciate this might not have been asked for before, but I imagine most users are not paying enough attention to notice if their screen sleeps - Then if they do realise, they have too specifically try and pinpoint which program is causing it to finally arrive here to ask for such a feature.

With older displays I of course wouldn't care quite as much, but with OLEDs becoming more mainstream and the need for maintenance they have, I feel more people will begin asking for this.


I have tried to override it myself with:
```powercfg -requestsoverride process "C:\Program Files\RustDesk\rustdesk.exe" display```

but unfortunately it seems to just be ignored.

So if you have any further suggestions or would be willing to consider adding an "allow screen to sleep/timeout" option in the settings, it would be greatly appreciated.

**Reply by inkcmd** [reply]
my lg oled c2 is sleeping fine with rustdesk service in tray and connected to someone. what you want? to connect remotely to your pc with oled and let oled sleep when you are using your pc? something unusual. are you sure that problem is not in your windows? im using windows 10, turning display off 10 min timeout, no windows sleep.
upd: no, i misunderstood a topicstarter and reproduced it too.
but maybe it's not a bug, more like a feature request, and it's an open question: is it realizable?
upd2: anydesk really did the same.
maybe remote connection can't wake up sleeping monitor when connected and you are using remote computer, i dont know,  its just my thoughts, looks and may be like shadow rdp.
i ordered a little arduino board and radio sensor long ago but still didn't install it. it should turn screen black and turn off display completely after some time without movement, one guy did this, maybe its good. but its additional hardware+software.

**Reply by SilentConfig** [reply]
To clarify -
When I am connected to a client from my main pc that has an OLED screen, the OLED screen will never sleep. I don't care about the screen on the remote PC.

**Reply by rustdesk** [reply, marked as answer]
Minimize your window if you do not want "never sleep".
https://github.com/rustdesk/rustdesk/blob/ffc73f86a0d75cc3f3c0ac02a133ef8312c77c72/flutter/lib/desktop/pages/remote_page.dart#L220

**Reply by SilentConfig** [reply]
I appreciate the intention to help but I leave rust desk open to monitor a machine, when I am at my machine.

To remember to minimise a window every time I walk away and then have to remember to bring it back up when im back is just not going to happen reliably.

I'm not a programmer but I see no reason why this SHOULDNT or COULDNT be added into the settings of rustdesk quite easily? to the benefit of some & the detriment of none?

I would imagine it's possible to add a checkbox in the settings for AllowDisplaySleep.
and then updating remote_page.dart with something like:
```  @override
  void onAllowDisplaySleep() {
      WakelockPlus.disable();
    }
```

Perhaps im misunderstanding?

in the very short time this request has been active theres already another person who found it & I believe would also likely benefit from the change.

**Reply by rustdesk** [reply]
> I see no reason why this SHOULDNT or COULDNT be added into the settings of rustdesk quite easily? to the benefit of some & the detriment of none?

Lock for meaningless dicusssions like this.

BTW, PR is welcome.
